.github/workflows/run_cafe_sync.yml

```
@@ -0,0 +1,49 @@
+name: Cafe Sync
+
+on:
+  schedule:
+    - cron: "0 * * * *"
+  push:
+    branches:
+      - master
+    paths:
+      - scripts/process.mjs
+
+jobs:
+  sync:
+    name: Sync
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Node
+        uses: actions/setup-node@v1.4.1
+        with:
+          node-version: 12.x
+
+      - name: Configure yarn cache
+        uses: actions/cache@v1
+        id: yarn-cache
+        with:
+          path: node_modules/
+          key: yarn-1-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            yarn-
+
+      - name: Install dependencies
+        run: yarn install
+        if: steps.yarn-cache.outputs.cache-hit != 'true'
+
+      - name: Get data
+        run: yarn run postinstall
+
+      - name: Push new Cafe
+        uses: github-actions-x/commit@v2.6
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          commit-message: "Add new Cafe"
+          files: src/data.json
+          rebase: 'true'
+          name: yihong0618
+          email: zouzou0208@gmail.com
```

```suggestion
      - uses: EndBug/add-and-commit@v4
        with:
          message: 'chore: update source'
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

And I would suggest using this action instead :)

---

Cool, I think you are right, thanks.  That will be easily. So can I create one `json file` only store the commit-sha on source repo if it change we trigger the action commit and netlify will do the other job?

---

Exactly! I think a plain text file with the sha will do. You can run this line to get the latest sha

```bash
curl https://api.github.com/repos/ElaWorkshop/awesome-cn-cafe/branches/master --silent | grep "sha" | head -n1 > sha
```